.github/actions/pr-review/pr_review.py

```
@@ -536,26 +547,45 @@ def call_model(system: str, user: str, mode: str, phase: str, correlation: str)
+        "X-Client-Request-Id": str(uuid.uuid4()),
+    }
+    payload = build_llm_payload(model, system, user, mode)
+    for attempt in range(1, MODEL_CONNECTION_ATTEMPTS + 1):
+        try:
+            response = requests.post(api_url, headers=headers, json=payload, timeout=timeout)
+        except requests.Timeout as exc:
+            # A timeout is ambiguous: the provider may complete and bill it
+            # after this runner stops waiting, so it is deliberately never
+            # retried.
+            log_phase(phase, attempt=attempt, status="timeout", correlation=correlation)
+            raise ModelTimeout("provider timed out") from exc
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)